docs/troubleshooting-guide/sql-error-message.md

```diff
@@ -0,0 +1,199 @@
+2. Review for missing brackets or commas on the line that is failing in your SQL query.
+3. Review for common syntax errors that are [specific to your SQL dialect][sql-reference-guide].
+
+### Common SQL reference guides
+
+Before you start, open up the SQL reference guide for the SQL dialect that you’re using. We’ve linked to some of the most common ones here:
+
+- [MySQL](https://dev.mysql.com/doc/refman/8.0/en/language-structure.html)
+- [PostgreSQL](https://www.postgresql.org/docs/current/sql-syntax-lexical.html)
+- [Microsoft SQL Server](https://docs.microsoft.com/en-us/sql/t-sql/language-reference)
+- [Amazon Redshift](https://docs.aws.amazon.com/redshift/latest/dg/cm_chap_SQLCommandRef.html)
+- [Google Bigquery](https://cloud.google.com/bigquery/docs/reference/standard-sql/lexical)
+- [Snowflake](https://docs.snowflake.com/en/sql-reference/constructs.html)
+- [I don’t know what SQL dialect to use][how-to-find-sql-dialect].
+
+### Common SQL syntax errors
+
+What does your error message say?
+
+- [My column or table name is "not found" or "not recognized"][sql-error-not-found].
+- [My SQL "function does not exist”][sql-error-function-does-not-exist].
+
+If your error message is not listed above, search or ask the community on [Discourse][discourse].
+
+#### My column or table name is "not found" or "not recognized".
+
+If your SQL query contains [SQL variables][sql-variable-def] that look like `{% raw %}{{ variable }}{% endraw %}`, go to [Troubleshooting SQL variables][troubleshoot-sql-variables] first.
+
+**Steps:**
+
+1.  Review the **structure** section of the [reference guide][sql-reference-guide] for your SQL dialect.
+
+    - Are you using the correct quotation marks? For example:
+
+      - `SELECT 'column_name'`
+      - `SELECT "column_name"`
+      - ` SELECT {% raw %}``column_name``{% endraw %} `
+
+    - Are you using the correct path to columns and tables? For example:
+
+      - `FROM table_name`
+      - `FROM schema_name.table_name`
+      - `FROM database_name.schema_name.table_name`
+
+    - Is your column name a reserved word? For example:
+
+      [In PostgresSQL, 'users' is a reserved key word](https://www.postgresql.org/docs/current/sql-keywords-appendix.html).
+
+      - `SELECT users` will throw an error.
+      - `SELECT "users"` will run correctly.
+
+2.  Review the [data reference][data-reference] for the tables and columns in your query.
+
+    - Does the column name exist in the table?
+
+    - If the column name exists, but you can’t query the column from the SQL editor:
+      - Ask your Metabase admin if the column was re-named or removed on the database side.
+      - If you’re a Metabase admin, you may need to [run a sync][database-syncing] to refresh your data.
+
+3.  If you no longer have an error message, but your query results are incorrect, go to [Troubleshooting SQL logic][troubleshoot-sql-logic].
+4.  If you're still stuck, search or ask the community on [Discourse][discourse].
+
+**Causes:**
```

Comment:
Wonder if we should change to "Explanation". Also, is a list the best format for this information?